aea/skills/error/handlers.py

```diff
@@ -67,10 +68,13 @@ def send_unsupported_protocol(self, envelope: Envelope) -> None:
         """
         logger.warning("Unsupported protocol: {}".format(envelope.protocol_id))
         reply = DefaultMessage(
-            type=DefaultMessage.Type.ERROR,
-            error_code=DefaultMessage.ErrorCode.UNSUPPORTED_PROTOCOL,
+            dialogue_reference=("", ""),
```

Comment:
Note, @5A11 and @marcofavorito we now have these default arguments which are `dialogue_reference`, `message_id` and `target`. In this scenario this creates an unnecessary overhead. We might want to allow for protocol specifications to specify that these arguments are not included. Thoughts?

Reply:
Ideally, shouldn't the dialogue be the same as the incoming message? 
Some thoughts:
- we could move the dialogue system to `Envelope`
- `dialogue_message=0` and `target=-1` as default values for undefined dialogues.

Reply:
Not sure what you mean with `the dialogue be the same as the incoming message`. We don't keep track of the dialogue on the `Envelope` level as the `dialogue` is a concept only maintained by the agents, not by the peers which route the envelope. This also makes sense when thinking about encryption as you can easily encrypt all the message body but not the envelope fields.

Reply:
We start indexing messages at 1 to avoid negative indeces on target.

Reply:
The commented lines correspond to the reply to an envelope whose protocol id is not supported.
Let's say the sender sends two envelopes with a protocol not supported by the recipient. With this code, the recipient answers with two messages. How the sender can distinguish between the two replies? This is an argument to include some identification system for envelopes.

Reply:
Actually, on discussion with @5A11 we realise that the error handler starts a new high level interaction in the default protocol. So we should set the LHS dialogue reference. The dialogue reference will allow the receiver to tell the two messages apart. Envelopes are part of another 'meta' protocol, we should be careful about having dependencies between the two.